daily_milestone.go

```diff
@@ -1,29 +1,225 @@
-	headers := map[string]string{"PrivateToken": token}
-	url := "https://" + base_url + "/projects"
-	page := 1
-	for {
+	}
+	req.Header.Add("PRIVATE-TOKEN", token)
+	resp, err := client.Do(req)
+	if err != nil {
+		logger.Println(err)
+
+	}
+	respByte, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		logger.Println("fail to read response data")
 
 	}
 
+	json.Unmarshal(respByte, &project)
+	defer resp.Body.Close()
+	for _, p := range project {
+		if p.State != "closed" {
+
+			titleList := []string{"Title: ", p.Title, ", DueDate", p.DueDate}
+			y := strings.Join(titleList, " ")
+			list = append(list, y)
+		}
+	}
+
+	return list, nil
 }
 
-//func get_milestones(base_url string, token string, project_id string)[]string{}
-//func create_milestone_data(advance int) []string{}
-//func create_milestones(base_url string, token string, project_id string, milestones string) string{}
+// CreateMilestoneData is used to check the due date using the time package of python
+func createMilestoneData(advance int) []string {
+	today := time.Now().Local()
+	date := today.AddDate(0, 0, advance)
+	list := []string{}
+	/*for i := 0; i < advance; i++ {
+		date := today.AddDate(0, 0, i)                          // To Format to ISOFormat and it converts to string so can be used in list directly
+		datelist := []string{"Title: ", date, "due_date", date} // Was unable to get a map in a list so made this
+		y := strings.Join(datelist, ",")
+		list = append(list, y)
+	}*/
+	fmt.Println(date)
+	fmt.Println(list)
+	return list
+}
+
+func createMilestones(baseURL string, token string, projectID string, milestones []string) string {
+	strurl := []string{"https://", baseURL, "/projects/", projectID, "/milestones"}
+	url := strings.Join(strurl, "")
+	client := &http.Client{}
+	for _, m := range milestones {
+		mbyte := bytes.NewReader([]byte(m))
+		req, err := http.NewRequest("POST", url, mbyte)
+		if err != nil {
+			logger.Println(err)
+			break
+		}
+		req.Header.Add("PRIVATE-TOKEN", token)
+		resp, err := client.Do(req)
+		if err != nil {
+			logger.Println(err)
+			break
+		}
+		defer resp.Body.Close()
+	}
+	return ("Milestones Created: " + strings.Join(milestones, ""))
+}
 
 func main() {
-	var apiKey, apiBase, namespace, project string
-	var advance int
-	flag.StringVar(&apiKey, "token", " ", "Gitlab api key/token.")
-	flag.StringVar(&apiBase, "base_url", " ", "Gitlab api base url")
-	flag.StringVar(&namespace, "namespace", " ", "Namespace to use in Gitlab")
-	flag.StringVar(&project, "projectname", " ", "Project to use in Gitlab")
-	flag.IntVar(&advance, "Advance", 30, "Define timeframe to generate milestones in advance.")
-	flag.Parse()
+	// Declaring variables for flags
+	var Token, APIBase, Namespace, Project string
+	var Advance int
+	APIBase = ""
+	// Command Line Parsing Starts
+	flag.StringVar(&Token, "Token", "bVYFTJaYtgAZesSofKbq", "Gitlab api key/token.")
+	flag.StringVar(&APIBase, "baseURL", "dev.seetheprogress.eu", "Gitlab api base url")
```

Comment:
Why call it baseURL in the cli and the functions, but APIBase as variable within main.go. Feels inconsistent.

Reply:
changed it